drivers/dht/dht.c

```diff
@@ -37,7 +37,9 @@
 #define mp_hal_pin_od_high_dht mp_hal_pin_od_high
 #endif
 
-STATIC mp_obj_t dht_readinto(mp_obj_t pin_in, mp_obj_t buf_in) {
+STATIC mp_obj_t dht_readinto(mp_obj_t pin_in, mp_obj_t buf_in, mp_obj_t irq_lock) {
+    mp_uint_t irq_state=0;
```

Comment:
This does not follow the code style in the file. Put spaces around '='. Also there seems to be no need for the 0 initialization.

Reply:
unfortunately you have to :-( That is why I added it.

+1: ../../drivers/dht/dht.c: In function 'dht_readinto':
./xtirq.h:48:5: error: 'irq_state' may be used uninitialized in this function [-Werror=maybe-uninitialized]
     __asm__ volatile ("wsr %0, ps; rsync" : : "a" (ps));
     ^
../../drivers/dht/dht.c:41:15: note: 'irq_state' was declared here
     mp_uint_t irq_state;
               ^
cc1: all warnings being treated as errors

Reply:
Well, either I'm missing something or compiler is wrong about that but well, in any way, we want to make it happy.